File: topic_6a/dog.cpp

#include "dog.h"

#ifndef DOG_H
#define DOG_H

// set breed
Dog::Dog(const std::string &breed) : Pet(breed, "") {}

// set breed and name
Dog::Dog(const std::string &breed, const std::string &name) : Pet("Dog", breed, name) {}

std::string Dog::speak() const
{
    return "woof";
}

#endif

File: java_practice/pets/Dog.java
package java_practice.pets;

public class Dog extends Pet {
    public Dog(String name, int age, int weight) {
        super(name, age, weight);
    }

    public String speak() {
        return "Woof";
    }

    public String wagTail() {
        return "Wagging its tail quickly.";
    }
}
